packages/lodestar/src/sync/backfill/verify.ts

```diff
@@ -0,0 +1,39 @@
+import {allForks, CachedBeaconState} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {Root, allForks as allForkTypes, ssz} from "@chainsafe/lodestar-types";
+import {GENESIS_SLOT} from "../../../../params/lib";
+import {IBlsVerifier} from "../../chain/bls";
+import {BackfillSyncError, BackfillSyncErrorCode} from "./errors";
+
+export function verifyBlockSequence(
+  config: IBeaconConfig,
+  blocks: allForkTypes.SignedBeaconBlock[],
+  anchorRoot: Root
+): void {
+  let nextRoot: Root = anchorRoot;
+  for (const block of blocks.slice(0).reverse()) {
+    const blockRoot = config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message);
```

Comment:
How long does it take to hash a block? Now that the batch size is 64 for how long will this hang the event loop?

Reply:
Aren't we doing that in regular range sync as well?

Reply:
Right! Then I guess it's okay.